### ISSUE TITLE
Updates decoder to allow generation of 512x512 images

### ISSUE THREAD

**Issue by bes-dev** [sberbank-ai/ru-dalle#42]
Thanks for your excellent work!

So, as I investigated from the code, currently, your pipeline has a quality bottleneck: DALL-E can predict latent code of the VQVAE only 1024 tokens length. Vanilla VQVAE can decode this code to an image with a size of 256x256. But there are some tricks (described in my article MobileStyleGAN): instead of predicting the image in the pixel domain, we can predict discrete wavelet transform of the target image. DWT transforms image size of [3; H, W] to tensor [12; H/2, W/2] and there is inverse transform without accuracy drop. For the VQVAE, we can predict x2 larger images from the same latent codes size of 1024 tokens. 

I built a pipeline (based on knowledge distillation) to train a modified VQVAE decoder that uses this trick. I got a baseline checkpoint that allows you to generate images 512x512 using your pre-trained ruDALL-E model. My results are available as open-source project here (link to baseline checkpoint also available): https://github.com/bes-dev/vqvae_dwt_distiller.pytorch.

I think that results can be better if we can train the whole DALL-E end-to-end using my version of the decoder (currently, I didn't modify codebook to save compatibility with ruDALL-E checkpoint).

**Comment by shonenkov** [comment]
I added optional params `dwt` for using this modification of decoder https://github.com/sberbank-ai/ru-dalle/pull/43

Also I would like to research with upgrading encoding part of vae and train using full set of images. I suppose this upgrade can improve not only visual quality of decoded images, but and help better understanding encoded codebooks at the cost higher input image size  

@bes-dev Thanks a lot for your issue! I will be waiting for new great ideas and issues from you 🙏
